src/screens/Home/components/SignUp/index.tsx

```diff
@@ -26,46 +30,115 @@ enum UserFieldIds {
+        </div>
+        <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
+          <label htmlFor={UserFieldIds.EMAIL} className={`m-bottom-2 ${styles.label}`}>
+            {i18next.t('SignUp:email')}
+          </label>
+          <input
+            type="email"
+            {...register(UserFieldIds.EMAIL, {
+              required: i18next.t('SignUp:required') as string,
+              pattern: {
+                value: /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i,
+                message: i18next.t('SignUp:invalidFormat') as string
+              }
+            })}
+            className={styles.input}
+          />
+          {errors.email && <span className={styles.inputError}>{errors.email.message}</span>}
+        </div>
+        <div className={`column start m-bottom-3 ${styles.inputContainer}`}>
+          <label htmlFor={UserFieldIds.PASSWORD} className={`m-bottom-2 ${styles.label}`}>
+            Password
+          </label>
+          <input
+            type="password"
+            {...register(UserFieldIds.PASSWORD, {
+              required: i18next.t('SignUp:required') as string,
+              pattern: {
+                value: /^(?=.*[A-Z])(?=.*[a-z])(?=.*[0-9])(?=.*[^\w\s\d])?.+$/,
+                message: i18next.t('SignUp:validFormatPassword') as string
+              },
+              minLength: {
+                value: 8,
+                message: i18next.t('SignUp:minLenghtPassword') as string
+              }
+            })}
+            className={styles.input}
+          />
+          {errors.password && <span className={styles.inputError}>{errors.password.message}</span>}
+        </div>
+        <div className={`column start m-bottom-5 ${styles.inputContainer}`}>
+          <label htmlFor={UserFieldIds.PASSWORD_CONFIRMATION} className={`m-bottom-2 ${styles.label}`}>
+            {i18next.t('SignUp:passwordConfirmation')}
+          </label>
+          <input
+            type="password"
+            {...register(UserFieldIds.PASSWORD_CONFIRMATION, {
+              required: true,
+              validate: value =>
+                value === watch(UserFieldIds.PASSWORD) || (i18next.t('SignUp:noMatchPassword') as string)
+            })}
+            className={styles.input}
+          />
+          {errors.password_confirmation && (
+            <span className={styles.inputError}>{errors.password_confirmation.message}</span>
+          )}
+        </div>
+        <button type="submit" className={`m-bottom-3 ${styles.signUpButton}`}>
+          {mutation.isLoading ? <Loading /> : 'Sign Up'}
+        </button>
+        <div className={`m-bottom-3 ${styles.line}`} />
+        <button type="button" className={styles.signUpButtonSecondary} />
+      </form>
+      {
+        <span className={!mutation.data?.ok ? `${styles.error}` : `${styles.noError}`}>
+          response :{mutation.data?.problem}
```

Comment:
same here, and also you should handle what happens if for some reason mutation.data is undefined. It could technically happen and you should have a default message if it does.